### ISSUE TITLE
TypeError: cannot perform reduce with flexible type

### ISSUE THREAD

**Issue by szymon-datalions** [szymon-datalions/pyinterpolate#74]
Numpy's error, probably related to the fact that areal id is a string type. In this case dictionary which stores strings will be created and all operations will be performed on generated integers.

**Comment by szymon-datalions** [comment]
And error message:

```
TypeError                                 Traceback (most recent call last)
<ipython-input-22-887357b65d37> in <module>()
      1 reg_mod.fit(areal_data=areal_data_prepared,
      2             areal_step_size=step_size, max_areal_range=maximum_range,
----> 3             point_support_data=points_in_area, ranges=16, weighted_lags=True, store_models=False)

/Users/szymonmolinski/anaconda3/envs/pyintpk/lib/python3.7/site-packages/pyinterpolate/semivariance/semivariogram_deconvolution/regularize_semivariogram.py in fit(self, areal_data, areal_step_size, max_areal_range, point_support_data, ranges, weighted_lags, store_models)
    284         # Regularize model
    285         self.initial_regularized_model = self._regularize(self.experimental_semivariogram_of_areal_data,
--> 286                                                           self.initial_theoretical_model_of_areal_data)
    287 
    288         # Calculate d-stat

/Users/szymonmolinski/anaconda3/envs/pyintpk/lib/python3.7/site-packages/pyinterpolate/semivariance/semivariogram_deconvolution/regularize_semivariogram.py in _regularize(self, empirical_semivariance, semivariance_model)
    128         theoretically_regularized_model_values = areal_semivariance.regularize_semivariogram(
    129             empirical_semivariance=empirical_semivariance,
--> 130             theoretical_semivariance_model=semivariance_model)
    131 
    132         return theoretically_regularized_model_values[:, 1]

/Users/szymonmolinski/anaconda3/envs/pyintpk/lib/python3.7/site-packages/pyinterpolate/semivariance/areal_semivariance/areal_semivariance.py in regularize_semivariogram(self, within_block_semivariogram, between_blocks_semivariogram, empirical_semivariance, theoretical_semivariance_model)
    135 
    136         if not within_block_semivariogram:
--> 137             self.within_block_semivariogram = self.calculate_mean_semivariance_between_blocks()
    138         else:
    139             self.within_block_semivariogram = within_block_semivariogram

/Users/szymonmolinski/anaconda3/envs/pyintpk/lib/python3.7/site-packages/pyinterpolate/semivariance/areal_semivariance/areal_semivariance.py in calculate_mean_semivariance_between_blocks(self, distances)
    209         # Calc average semivariance
    210         avg_semivariance = calculate_average_semivariance(self.distances_between_blocks, self.inblock_semivariance,
--> 211                                                           self.areal_lags, self.areal_ss)
    212         return avg_semivariance
    213 

/Users/szymonmolinski/anaconda3/envs/pyintpk/lib/python3.7/site-packages/pyinterpolate/semivariance/areal_semivariance/within_block_semivariance/calculate_average_semivariance.py in calculate_average_semivariance(between_block_distances, semivariance_within_block_points, lags, step_size)
     86             avg_sem = 0
     87         else:
---> 88             avg_sem = 0.5 * (np.mean(avg_sem))
     89 
     90         # Append average semivariance for given lag

<__array_function__ internals> in mean(*args, **kwargs)

/Users/szymonmolinski/anaconda3/envs/pyintpk/lib/python3.7/site-packages/numpy/core/fromnumeric.py in mean(a, axis, dtype, out, keepdims)
   3333 
   3334     return _methods._mean(a, axis=axis, dtype=dtype,
-> 3335                           out=out, **kwargs)
   3336 
   3337 

/Users/szymonmolinski/anaconda3/envs/pyintpk/lib/python3.7/site-packages/numpy/core/_methods.py in _mean(a, axis, dtype, out, keepdims)
    149             is_float16_result = True
    150 
--> 151     ret = umr_sum(arr, axis, dtype, out, keepdims)
    152     if isinstance(ret, mu.ndarray):
    153         ret = um.true_divide(

TypeError: cannot perform reduce with flexible type
```

**Comment by szymon-datalions** [comment]
Bug fixed in the package version 0.2.3
